cpp/src/randomforest/randomforest.cu

```diff
@@ -133,7 +132,8 @@ void preprocess_labels(int n_rows, std::vector<int>& labels,
  */
 void postprocess_labels(int n_rows, std::vector<int>& labels,
                         std::map<int, int>& labels_map, bool verbose) {
-  if (verbose) std::cout << "Postrocessing labels\n";
+  ML::Logger::get().setLevel(verbose ? CUML_LEVEL_DEBUG : CUML_LEVEL_INFO);
```

Comment:
I'm wondering if it would make more sense to remove the verbose arguments altogether and just have the prims log to the relevant level. That way we can set the logger to the desired level in Python and start having the Python layer support levels (where perhaps a boolean could mean set to `INFO` or `DEBUG`) This would also remove the need to dynamically set the log level in each and every prim/algorithm while also removing the need for a verbose option. I'm mostly concerned that this will become hard to maintain and setting the log level in one place at the very beginning will allow log to the appropriate level anywhere in C++ without having to propagate the verbose option down. What do you think?

Reply:
An ideal solution here is for us to expose `verbose` not as a flag, but as an integer. But that would mean it will require an interface change both in c++ and python. That's why I hesitated to make that change in this PR. I'd like to address this in a separate PR (and also will file an issue on this too), if you agree.

Reply:
I do agree that we should expose a verbose argument on the C++ side as an integer that can set the exact level, rather than a boolean. On the Python side, I think we could continue to give the user an option of boolean (false = WARN, true = ACTIVE_LEVEL) or integer. 

I would like to minimize the amount of places where `setLevel` is being called explicitly, especially while everything appears to be sharing the same underlying logger. Mostly, I'm concerned that with different functions making their own subsequent calls to `setLogger`, debugging could become a really confusing process, which could result in expected log statements not being printed because a function we're calling modified the level. For this reason, I suggest that we limit `setLevel` only the wrapper functions called  by Python. This would still allow us to set the log level in tests and benchmarks, but the prims themselves would not be setting it, nor would they need to expose their own `verbose` argument. What do you think?

Reply:
I don't have any issues with adding these commits to this PR itself, as long as you're OK reviewing them altogether.

Reply:
Sounds good to me.

Reply:
Cool. Will send updates today.

Reply:
To demonstrate this, I have submitted a few commits just updating dbscan algo for handling this verbosity. If this looks good to you, I'll go ahead and update others too.